cabotage/celery/tasks/build.py

```diff
@@ -390,45 +391,51 @@ def auth(dxf, response):
 def _fetch_github_file(
     github_repository="owner/repo", ref="main", access_token=None, filename="Dockerfile"
 ):
-    headers = {
-        "Accept": "application/vnd.github+json",
-        "X-GitHub-Api-Version": "2022-11-28",
-    }
-    if access_token is not None:
-        headers["Authorization"] = f"token {access_token}"
-    response = requests.get(
-        f"https://api.github.com/repos/{github_repository}/contents/{filename}",
-        params={"ref": ref},
-        headers=headers,
-    )
-    if response.status_code == 404:
+    g = Github(access_token)
+    try:
+        content_file = g.get_repo(github_repository).get_contents(filename, ref=ref)
+        if content_file.encoding == "base64":
+            return b64decode(content_file.content).decode()
+        return content_file.content
+    except UnknownObjectException:
         return None
-    if response.status_code == 200:
-        data = response.json()
-        if data["encoding"] == "base64":
-            return b64decode(response.json()["content"]).decode()
-    response.raise_for_status()
+
+
+def _is_imposter_commit(
+    github_repository="owner/repo", ref=None, sha=None, access_token=None
```

Comment:
Nit: maybe make `ref` a required kwargs?

From a quick look `None` will only produce an error state:

```suggestion
    github_repository="owner/repo", *, ref, sha, access_token=None
```

Reply:
Nice, TIL PEP-3102